arch/arc/include/asm/spinlock.h

```diff
@@ -239,16 +239,27 @@ static inline void arch_write_unlock(arch_rwlock_t *rw)
 	"4: ; --- done ---			\n"				\
 
 #define SCOND_FAIL_RETRY_VARS							\
-	  ,[delay] "+&r" (delay), [tmp] "=&r"	(tmp)				\
+	  ,[delay] "+&r" (delay), [wait] "+&r" (wait), [tmp] "=&r" (tmp)	\
+
+#define SPINLOCK_WAIT_LOOP_WITH_DELAY						\
+	"1:	llock	%[val], [%[slock]]	\n"				\
+	"	brne	%[val], %[LOCKED], 11f	\n"	/* if LOCKED */		\
```

Comment:
/* If not LOCKED */

Reply:
I've reviewed it and think we can merge this.

Reply:
Resolved in new rebase, thank you.